pennylane/_qubit_device.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2018-2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :class:`QubitDevice` abstract base class.
+"""
+# pylint: disable=too-many-format-args
+import abc
+
+import numpy as np
+import itertools
+from collections import OrderedDict
+
+import pennylane as qml
+from pennylane.operation import Operation, Observable, Sample, Variance, Expectation, Probability, Tensor
+from pennylane.qnodes import QuantumFunctionError
+from pennylane import Device, DeviceError
+
+
+class QubitDevice(Device):
+    """Abstract base class for PennyLane qubit devices.
+
+    Args:
+        wires (int): number of subsystems in the quantum state represented by the device.
+            Default 1 if not specified.
+        shots (int): Number of circuit evaluations/random samples used to estimate
+            expectation values of observables. Defaults to 1000 if not specified.
+    """
+    #pylint: disable=too-many-public-methods
+    def __init__(self, wires=1, shots=1000, analytic=False):
```

Comment:
How come analytic is set to false by default? Is that what is done in `Device`?

Reply:
Nope, this was explicit here, will change it to ``True``, thanks!